docs/API.md

```diff
@@ -963,48 +967,94 @@ __Return Value__
 __Example__
 
 ```py
-import time
 from datetime import datetime
-from minio import CopyConditions
+from minio.commonconfig import CopySource
```

Comment:
This example is confusing. It looks like it is a complete test script example, but actually the intent is not to provide a full example here as in the other api commands. 
The reason for the confusion is, I think, putting everything in the same presentation block.
Separating each excerpt in its unique presentation block with a one sentence explanation/comment is going to solve the problem:

The following shows the required module(s) for `copy_object` api to work, that is `minio.commonconfig`.
 `datetime` module is only needed if a time related conditional arg, like `modified_since` or `unmodified_since`, is used. So, It is better to show them together with the "modified condition"ed copy object examples.
```
from minio import Minio
# copy_object api needs minio.commonconfig module to be imported
from minio.commonconfig import CopySource

client = Minio(
    "s3.amazonaws.com",
    access_key="YOUR-ACCESSKEY",
    secret_key="YOUR-SECRETKEY",
)

# Copy object "my-source-bucketname/my-source-objectname" to "my-bucketname/my-objectname"
client.copy_object(
    "my-bucketname",
    "my-objectname",
    CopySource("my-source-bucketname", "my-source-objectname"),
)
```
```
# Copy object "my-source-bucketname/my-source-objectname", if its version id matches the provided
# version id 
:
:
client.copy_object(
    "my-bucketname",
    "my-objectname",
    CopySource(
        "my-source-bucketname", "my-source-objectname",
        version_id="b6602757-7c9c-449b-937f-fed504d04c94",
    ),
)
```
```
# Copy object "my-source-bucketname/my-source-objectname", if its "modified-since" 
# value is later than or equal to the provided modified date and time and its etag value matches the
# provided etag value.
# Print the object name if the object is copied over successfully,
# otherwise an error message about the failed condition.
:
from datetime import datetime, timezone
:
:
result = client.copy_object(
    "my-bucketname",
    "my-objectname",
    CopySource(
        "my-source-bucketname",
        "my-source-objectname",
        match_etag="31624deb84149d2f8ef9c385918b653a",
        modified_since=datetime(2014, 4, 0, tzinfo=timezone.utc),
    ),
)
print(result.object_name, result.version_id)
```
```
# Copy object "my-source-bucketname/my-source-objectname", and
# replace its meta data with the provided meta data (key: value) pair.
# Add meta data (key: value) pair, if there is no meta data with the provided key name, `X-Amz-Meta-<key>`
:
from minio.commonconfig import REPLACE, CopySource
:
:
metadata = {"test_meta_key": "test_meta_value"}
result = client.copy_object(
    "my-bucket",
    "my-object",
    CopySource("my-sourcebucket", "my-sourceobject"),
    metadata=metadata,
    metadata_directive=REPLACE,
)
```

Reply:
I am copying  code from `examples/copy_object.py` to avoid confusion. Hope this helps.